dotnet/src/InternalUtilities/src/Http/HttpRequest.cs

```diff
@@ -42,12 +41,7 @@ internal static class HttpRequest
         HttpContent? content = null;
         if (payload is not null)
         {
-            byte[] utf8Bytes = payload is string s ?
-                Encoding.UTF8.GetBytes(s) :
-                JsonSerializer.SerializeToUtf8Bytes(payload, JsonOptionsCache.Default);
-
-            content = new ByteArrayContent(utf8Bytes);
-            content.Headers.ContentType = new MediaTypeHeaderValue("application/json") { CharSet = "utf-8" };
+            return new StringContent(JsonSerializer.Serialize(payload, JsonOptionsCache.Default), Encoding.UTF8, "application/json");
```

Comment:
Using the ByteArrayContent broke the usage of this method when consuming streaming apis from the Hugging face Inference API.

Using the StringContent, both the streaming and non-streaming text worked just fine.

Reply:
There's no reason it should have broken it: StringContent _is_ a ByteArrayContent that just turns the string intuitive bytes. Before you make this change, it'd be good to understand what exactly the problem is and fix that. Using StringContent here is more expensive.

Reply:
@stephentoub When using the current code, the code holds at the https://github.com/microsoft/semantic-kernel/blob/bc2d83c11e88107cb7eab11a3629002f4a5dccf6/dotnet/src/InternalUtilities/src/Http/HttpClientExtensions.cs#L29 without moving to the next line.

Changing the Content to StringContent, worked out.

Reply:
What's the input to this method (type and value). that's failing?

Reply:
Will work in a reproduction scenario for this and share soon.

Reply:
Wasn't able to reproduce the problem again in my local environment, closing this.

Reply:
I'll look at it when I'm back at my computer. It's possible there's an issue here, but if there is I'd still like to go through ByteArrayContent to avoid unnecessarily producing an intermediate stream.